docs/guide/migration-from-0.6-to-1.0.md

```diff
@@ -0,0 +1,199 @@
+  licenseKey: 'agpl-v3',
+}
+```
+
+After:
+```js
+const options = {
+  // use `gpl-v3` instead of `agpl-v3`
+  licenseKey: 'gpl-v3',
+}
+```
+
+### Free non-commercial license
+
+If you use the free non-commercial license, switch to the GPLv3 license or purchase a commercial license.
+
+### Commercial license
+
+If you use a commercial license, you don't need to make any changes.
+
+## Step 2: Change `sheetName` to `sheetId`
+
+Most sheet-related methods now take the `sheetID` number parameter instead of the `sheetName` string parameter.
+
+For example, use the `clearSheet()` method in this way:
+
+Before:
+```js
+const hfInstance = HyperFormula.buildFromSheets({
+ MySheet1: [ ['=SUM(MySheet2!A1:A2)'] ],
+ MySheet2: [ ['10'] ],
+});
+
+const changes = hfInstance.clearSheet('MySheet2');
+```
+
+After:
+```js
+const hfInstance = HyperFormula.buildFromSheets({
+ MySheet1: [ ['=SUM(MySheet2!A1:A2)'] ],
+ MySheet2: [ ['10'] ],
+});
+
+// use `sheetId` instead of `sheetName`
+const changes = hfInstance.clearSheet(0);
+```
+
+The only methods still accepting the `sheetName` string parameter are now:
+- `addSheet()`
+- `isItPossibleToAddSheet()`
+- `doesSheetExist()`
+- `getSheetId()`
+
+Also, these methods still accept the `newName` string parameter:
+- `renameSheet()`
+- `isItPossibleToRenameSheet()`
+
+## Step 3: Adapt to the `matrix`->`array` name changes
+
+Adapt to the following changes in configuration option names, API method names and exception names:
+
+### Configuration option names
+
+| Before                  | After                    |
+|-------------------------|--------------------------|
+| `matrixColumnSeparator` | `arrayColumnSeparator`   |
+| `matrixRowSeparator`    | `arrayRowSeparator`      |
+
+### API method names
+
+| Before               | After               |
+|----------------------|---------------------|
+| `matrixMapping`      | `arrrayMapping`     |
+| `isCellPartOfMatrix` | `isCellPartOfArray` |
+
+### Exception names
+
+| Before                         | After                         |
+|--------------------------------|-------------------------------|
+| `SourceLocationHasMatrixError` | `SourceLocationHasArrayError` |
+| `TargetLocationHasMatrixError` | `TargetLocationHasArrayError` |
+
+
+## Step 4: Drop the matrix formula notation
+
+Switch from the matrix formula notation to the array formula notation.
+
+For more information on the array formula notation, go [here](arrays.md).
+
+Before:
+```js
+={ISEVEN(A2:A5*10)}
+```
+
+After:
+```js
+// when the `useArrayArithmetic` configuration option is set to `false`
+// use the ARRAYFORMULA function
+=ARRAYFORMULA(ISEVEN(A2:A5*10))
+
+// when the `useArrayArithmetic` configuration option is set to `true`
+// you don't need to use the ARRAYFORMULA function
+=(ISEVEN(A2:A5*10))
+```
+
+## Step 5: Drop the `matrixDetection` and `matrixDetectionThreshold` options
+
+Remove the `matrixDetection` and `matrixDetectionThreshold` options from your HyperFormula configuration.
```

Comment:
Hmm, I don't have an explanation for this one. @izulin maybe? We now do this automatically, and no options are necessary for optimizations, right?